var/spack/repos/builtin/packages/libxml2/package.py

```diff
@@ -252,6 +254,11 @@ def configure_args(self):
         else:
             args.append("--without-python")
 
+        if spec.satisfies("+http"):
+            args.append("--with-http")
+        else:
+            args.append("--without-http")
```

Comment:
Recommend using the helper (https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#with-or-without):

```suggestion
        args.append(self.with_or_without("http"))
```